Sources/PackageCollections/Model/Collection.swift

```diff
@@ -71,52 +71,57 @@ extension PackageCollectionsModel {
 
 extension PackageCollectionsModel {
     /// Represents the source of a `PackageCollection`
-    public enum PackageCollectionSource: Equatable {
-        /// Package feed at URL
-        case feed(URL)
+    public struct CollectionSource: Equatable, Hashable, Codable {
```

Comment:
@yim-lee please note this model change. need to create a map of source-type -> provider so that we can add more providers per type in the future, this meant I had to change source from enum to struct with type, etc

Reply:
My main concern is whether all sources would have `url` and I think it's safe to assume that. 🤔

Reply:
yes I debated the same thing and decided its probably a safe assumption, at least for now

Reply:
👍